src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -536,6 +541,13 @@ async def determine_finalized_setup_kwargs(request: GenerateSetupPyRequest) -> F
     resolved_setup_kwargs = await Get(SetupKwargs, ExportedTarget, exported_target)
     setup_kwargs = resolved_setup_kwargs.kwargs.copy()
 
+    # Check interpreter constraints
+    if len(request.interpreter_constraints) > 1:
+        raise SetupPyError(
+            f"Expected a single set of interpreter constraints for {target.address}, "
+            f"got: {request.interpreter_constraints}."
+        )
```

Comment:
Can add test for this check, given that it is a check that makes sense.. ?

Reply:
The check does make sense. When given a list of ICs, Pants (and Pex) inserts a logical `OR` between them in some operations. Perhaps only when searching for compatible interpreters on the local system IIRC. That "feature" should be dropped, but is still laying around. It has zero support in the PyPA ecosystem though and cannot be represented there where there is only the `,` operator in an individual IC for logical `AND`. So anyone with multiple ICs that wants to publish a dist needs to convert those into a single IC by using exclusions. For example: [">=2.7,<3", ">=3.5,<3.11"] (Pex's own constraints), should be converted by the user to ["">=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,<3.11"]. If you're in the mood to write books here, that's some seed material.